sts_synindex_external.R

```diff
@@ -118,43 +132,20 @@ if (nrow(synapse_fileview)>0) {
         dplyr::select(parent = parentId,
                       s3_file_key = dataFileKey,
                       md5_hash = dataFileMD5Hex))
-  } else {
-    synapse_manifest_to_upload <- synapse_manifest
-  }
-
-## Index in Synapse
-## For each file index it in Synapse given a parent synapse folder
-if(nrow(synapse_manifest_to_upload) > 0){ # there are some files to upload
-  for(file_number in seq(nrow(synapse_manifest_to_upload))){
-
-    # file and related synapse parent id
-    file_= synapse_manifest_to_upload$path[file_number]
-    parent_id = synapse_manifest_to_upload$parent[file_number]
-    s3_file_key = synapse_manifest_to_upload$s3_file_key[file_number]
-    # this would be the location of the file in the S3 bucket, in the local it is at {AWS_PARQUET_DOWNLOAD_LOCATION}
-
-    absolute_file_path <- tools::file_path_as_absolute(file_) # local absolute path
-
-    temp_syn_obj <- synapser::synCreateExternalS3FileHandle(
-      bucket_name = "recover-main-project",
-      s3_file_key = s3_file_key, #
-      file_path = absolute_file_path,
-      parent = parent_id
-    )
-
-    # synapse does not accept ':' (colon) in filenames, so replacing it with '_colon_'
-    new_fileName <- stringr::str_replace_all(temp_syn_obj$fileName, ':', '_colon_')
-
-    f <- File(dataFileHandleId=temp_syn_obj$id,
-              parentId=parent_id,
-              name = new_fileName) ## set the new file name
-
-    f <- synStore(f)
-
-  }
+  synapse_manifest_to_upload <- 
+    synapse_manifest_to_upload %>% 
+    mutate(file_key = gsub("cohort_", "cohort=", file_key),
+           s3_file_key = gsub("cohort_", "cohort=", s3_file_key))
+} else {
+  synapse_manifest_to_upload <- 
+    synapse_manifest %>% 
+    mutate(file_key = gsub("cohort_", "cohort=", file_key),
+           s3_file_key = gsub("cohort_", "cohort=", s3_file_key))
 }
 
-# ## Upload the contents of the parquet_final folder to Synapse
-# for (i in seq_along(synapse_manifest$path)) {
-#   synStore(File(synapse_manifest$path[i], parent=synapse_manifest$parent[i]))
-# }
+# Upload the contents of the parquet_final folder to Synapse
+system.time(
+  for (i in seq_along(synapse_manifest_to_upload$path)) {
+    synStore(File(synapse_manifest_to_upload$path[i], parent=synapse_manifest_to_upload$parent[i]))
+  }
+)
```

Comment:
Ignore these line: this was the old method for storing the final external parquet datasets and is from old commits. The updated sync logic (archive + current) is in PR #2.